fastly/data_source_fastly_tls_configuration_ids.go

```diff
@@ -0,0 +1,44 @@
+package fastly
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/hashcode"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceFastlyTLSConfigurationIDs() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceFastlyTLSConfigurationIDsRead,
+		Schema: map[string]*schema.Schema{
+			"ids": {
+				Type:        schema.TypeList,
+				Description: "List of IDs corresponding to available TLS configurations.",
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func dataSourceFastlyTLSConfigurationIDsRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*FastlyClient).conn
+
+	configurations, err := listTLSConfigurations(conn)
+	if err != nil {
+		return err
+	}
+
+	var ids []string
+	for _, configuration := range configurations {
+		ids = append(ids, configuration.ID)
+	}
+
+	// 2.x upgrade note - `hashcode.String` was removed from the SDK
+	// Code will need to be copied into this repository
+	// https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#removal-of-helper-hashcode-package
+	d.SetId(fmt.Sprintf("%d", hashcode.String(""))) // hashCode should include any filters set when they are added
```

Comment:
If I recall correctly, this was mainly as a placeholder anticipating the addition of filters to this data source at some point, similarly to how the [`aws_ami_ids`](https://github.com/hashicorp/terraform-provider-aws/blob/bc48c2178fff1c5b1b142481069e45fcb5870bd5/aws/data_source_aws_ami_ids.go#L111) data source does it, but in this case there are no filters so an empty string was used in the mean time. It could be hardcoded to `0` if you think that would be cleaner, and a `hashcode` or similar could be added in at a later point if additional filters are added?

Reply:
> anticipating the addition of filters to this data source at some point

if it's likely we're going to see filters come in the future, then I think this is fine to leave 'as is', otherwise I'd suggest we set some kind of value to be hashed.